tests/bofire/strategies/test_enting.py

```diff
@@ -0,0 +1,234 @@
+from entmoot import EntingParams
+from entmoot.problem_config import FeatureType, ProblemConfig
+
+import bofire.data_models.strategies.api as data_models
+from bofire.benchmarks.api import Hartmann
+from bofire.data_models.constraints.api import (
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+)
+from bofire.data_models.domain.api import Domain
+from bofire.data_models.features.api import (
+    CategoricalInput,
+    ContinuousInput,
+    ContinuousOutput,
+    DiscreteInput,
+    Input,
+)
+from bofire.data_models.objectives.api import MaximizeObjective, MinimizeObjective
+from bofire.strategies.api import EntingStrategy
+from bofire.strategies.predictives.enting import domain_to_problem_config
+from tests.bofire.strategies.test_base import domains
+
+
+@pytest.fixture
+def common_args():
+    return {
+        "dist_metric": "l1",
+        "acq_sense": "exploration",
+        "solver_name": "gurobi",
+        "solver_verbose": False,
+    }
+
+
+def test_enting_not_fitted(common_args):
+    data_model = data_models.EntingStrategy(domain=domains[0], **common_args)
+    strategy = EntingStrategy(data_model=data_model)
+
+    msg = "Uncertainty model needs fit function call before it can predict."
+    with pytest.raises(AssertionError, match=msg):
+        strategy._ask(1)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        {"acq_sense": "penalty", "beta": 0.1, "dist_metric": "l2", "max_depth": 3},
+    ],
+)
+def test_enting_param_consistency(params):
+    # compare EntingParams objects between entmoot and bofire
+    data_model = data_models.EntingStrategy(domain=domains[0], **params)
+    enting_params = EntingParams(**data_model.dump_enting_params())
+    assert all(
+        (
+            enting_params.unc_params.acq_sense == params["acq_sense"],
+            enting_params.unc_params.beta == params["beta"],
+            enting_params.unc_params.dist_metric == params["dist_metric"],
+            enting_params.tree_train_params.train_params.max_depth
+            == params["max_depth"],
+        )
+    )
+
+    # check that the parameters propagate to the model correctly
+    strategy = EntingStrategy(data_model=data_model)
+    assert strategy._enting._acq_sense == data_model.acq_sense
+    assert strategy._enting._beta == data_model.beta
+
+
+@pytest.mark.parametrize(
+    "allowed_k",
+    [1, 3, 5, 6],
+)
+@pytest.mark.slow
+def test_nchoosek_constraint_with_enting(common_args, allowed_k):
+    benchmark = Hartmann(6, allowed_k=allowed_k)
+    samples = benchmark.domain.inputs.sample(10, seed=43)
+    experiments = benchmark.f(samples, return_complete=True)
+
+    data_model = data_models.EntingStrategy(domain=benchmark.domain, **common_args)
+    strategy = EntingStrategy(data_model)
+
+    strategy.tell(experiments)
+    proposal = strategy.ask(1)
+
+    input_values = proposal[benchmark.domain.get_feature_keys(Input)]
+    assert (input_values != 0).sum().sum() <= allowed_k
+
+
+@pytest.mark.slow
```

Comment:
This makes sense, could you check if gurobi is installed in somehow the same way that you check that entmoot is installed and then add another `skipif`. This would be a bit cleaner. If not possible, then we keep it.